index.ts

```diff
@@ -58,82 +58,126 @@ export interface ServiceDetails<S> {
+    const exposed = this.serviceDetails.expose;
+    const methods = exposed.map((m) => m.methodName);
+    const serviceName = this.serviceDetails.service;
+    const serviceHelp = this.serviceDetails.help || serviceName + " service";
+
+    const meta = {
+      serviceName,
+      multiArg: false,
+      help: serviceHelp,
+      interfaces: exposed.map((method: MethodDetails<S>) => {
+        if (typeof method === "string") {
+          throw new Error(
+            "Schema for expose has changed. Please refer to @loke/http-rpc documentation."
+          );
+        }
+        const {
+          methodName,
+          methodTimeout = 60000,
+          help,
+          paramNames = [],
+        } = method;
+
+        return {
+          methodName,
+          paramNames,
+          methodTimeout,
+          help: help || methodName + " method",
+        };
+      }),
+    };
+
+    return (
+      req: { path: string; method: string; body: unknown },
+      res: { json: (body: unknown) => void },
+      next: (err?: Error) => void
+    ) => {
+      const methodName: string = req.path.slice(1); // remove leading slash
+
+      if (req.method === "GET" && req.path === "/") {
+        return res.json(meta);
+      }
+
+      if (req.method === "GET" && methods.includes(methodName)) {
+        return res.json(
+          meta.interfaces.find((i) => i.methodName === methodName)
         );
       }
-      const {
-        methodName,
-        methodTimeout = 60000,
-        help,
-        paramNames = [],
-      } = method;
-
-      return {
-        methodName,
-        paramNames,
-        methodTimeout,
-        help: help || methodName + " method",
-      };
-    }),
-  };
 
-  return (
-    req: { path: string; method: string; body: unknown },
-    res: { json: (body: unknown) => void },
-    next: (err?: Error) => void
-  ) => {
-    const methodName: string = req.path.slice(1); // remove leading slash
+      if (req.method !== "POST" || !methods.includes(methodName)) {
+        return next();
+      }
 
-    if (req.method === "GET" && req.path === "/") {
-      return res.json(meta);
-    }
+      const requestMeta = { handler: `${serviceName}.${methodName}` };
+      const end = requestDuration.startTimer(requestMeta);
 
-    if (req.method === "GET" && methods.includes(methodName)) {
-      return res.json(meta.interfaces.find((i) => i.methodName === methodName));
-    }
+      requestCount.inc(requestMeta);
 
-    if (req.method !== "POST" || !methods.includes(methodName)) {
-      return next();
-    }
+      return Promise.resolve()
+        .then(() => this.service[methodName].call(this.service, req.body))
+        .then((result) => (console.log("result", result), res.json(result)))
+        .catch((err) => {
+          failureCount.inc(Object.assign({ type: err.type }, requestMeta));
+          next(new RpcError(serviceName, methodName, err));
+        })
+        .finally(end);
+    };
+  }
+
+  toWellKnownMeta() {
```

Comment:
Again, doc comments for anything that is public

```suggestion
  /**
   * ... add description here
   */
  toWellKnownMeta() {
```

Reply:
Adding comments now